tests/general/ArchiveOrgBearTest.py

```diff
@@ -0,0 +1,322 @@
+    :param override_head: True means the url's head request will be overriden
+                          by this method.
+    """
+    if override_head:
+        m.head(url, headers={})
+    if is_archived:
+        m.head(
+            'http://timetravel.mementoweb.org/timegate/%s' % url,
+            headers={
+                'Vary': 'Accept-Datetime', 'Location': 'http:/'
+                        '/web.archive.org/web/20170421070937/h'
+                        'ttps://www.google.com/',
+                'Link': '<ht'
+                        'tp://google.com>;rel="original" ,<htt'
+                        'p://timetravel.mementoweb.org/timemap'
+                        '/link/http://google.com>;rel="timemap'
+                        '"; type="application/link-format",<ht'
+                        'tp://web.archive.org/web/201704210706'
+                        '53/http://www.google.com/>;rel="memen'
+                        'to last"; datetime="Fri, 21 Apr 2017 '
+                        '07:06:53 GMT",<http://web.archive.bib'
+                        'alex.org:80/web/19981111184551/http:/'
+                        '/google.com/>;rel="memento first"; da'
+                        'tetime="Wed, 11 Nov 1998 18:45:51 GMT'
+                        '"'
+            },
+            status_code=302)
+        m.head(
+            'http://web.archive.org/web/20170421070937/'
+            'https://www.google.com/',
+            headers={
+                'Link': '<https://www.google.com/>; rel="origi'
+                        'nal", <http://web.archive.org/web/tim'
+                        'emap/link/https://www.google.com/>; r'
+                        'el="timemap"; type="application/link-'
+                        'format", <http://web.archive.org/web/'
+                        'https://www.google.com/>; rel="timega'
+                        'te", <http://web.archive.org/web/1998'
+                        '1111184551/http://google.com:80/>; re'
+                        'l="first memento"; datetime="Wed, 11 '
+                        'Nov 1998 18:45:51 GMT", <http://web.a'
+                        'rchive.org/web/20170420064646/http://'
+                        'www.google.com/>; rel="prev memento";'
+                        ' datetime="Thu, 20 Apr 2017 06:46:46 '
+                        'GMT", <http://web.archive.org/web/201'
+                        '70421070937/https://www.google.com/>;'
+                        ' rel="memento"; datetime="Fri, 21 Apr'
+                        ' 2017 07:09:37 GMT", <http://web.arch'
+                        'ive.org/web/20170422071758/https://ww'
+                        'w.google.com/>; rel="next memento"; d'
+                        'atetime="Sat, 22 Apr 2017 07:17:58 GM'
+                        'T", <http://web.archive.org/web/19981'
+                        '111184551/http://google.com:80/>; rel'
+                        '="last memento"; datetime="Wed, 11 No'
+                        'v 1998 18:45:51 GMT"'
```

Comment:
I think it'd be good to break the lines a bit more readable, i.e. after the semicolons where possible. Or maybe put them into a file?